libs/shared/flagd-core/src/lib/parser.ts

```diff
@@ -14,31 +25,56 @@ const errorMessages = 'invalid flagd flag configuration';
 
 /**
  * Validate and parse flag configurations.
+ * @param flagConfig The flag configuration string.
+ * @param strictValidation Validates against the flag and targeting schemas.
+ * @param logger The logger to be used for troubleshooting.
+ * @returns The parsed flag configurations.
  */
-export function parse(flagCfg: string, throwIfSchemaInvalid: boolean, logger?: Logger): Map<string, FeatureFlag> {
+export function parse(flagConfig: string, strictValidation: boolean, logger: Logger): FlagSet {
   try {
-    const transformed = transform(flagCfg);
-    const flags: { flags: { [key: string]: Flag } } = JSON.parse(transformed);
-    const isValid = validate(flags);
+    const transformed = transform(flagConfig);
+    const parsedFlagConfig: FlagConfig = JSON.parse(transformed);
+
+    const isValid = validate(parsedFlagConfig);
     if (!isValid) {
       const message = `${errorMessages}: ${JSON.stringify(validate.errors, undefined, 2)}`;
-      logger?.warn(message);
-      if (throwIfSchemaInvalid) {
+      if (strictValidation) {
         throw new ParseError(message);
+      } else {
+        logger.debug(message);
       }
     }
     const flagMap = new Map<string, FeatureFlag>();
 
-    for (const flagsKey in flags.flags) {
-      flagMap.set(flagsKey, new FeatureFlag(flags.flags[flagsKey]));
+    const flagSetMetadata = parsedFlagConfig.metadata ?? {};
+
+    for (const flagsKey in parsedFlagConfig.flags) {
+      const flag = parsedFlagConfig.flags[flagsKey];
+      flagMap.set(
+        flagsKey,
+        new FeatureFlag(
+          flagsKey,
+          {
+            ...flag,
+            metadata: {
```

Comment:
It may be worth adding a comment here as well about flag metadata taking priority.